### DISCUSSION TITLE
What happened to the XPU backend?

### DISCUSSION THREAD

**Discussion by dionhaefner** [jax-ml/jax#24483] [Q&A] [answered]
In mpi4jax, we noticed that it's not possible to register XPU extensions anymore. Was that backend cut from JAX?

```
NotImplementedError: Registering an MLIR lowering rule for primitive sendrecv_mpi for an unknown platform xpu. Known platforms are: METAL, cpu, cuda, gpu, rocm, tpu.
```

**Comment by hawkinsp** [comment, marked as answer]
We've never had an in-tree XPU backend.

We added checking code that means you can't register lowering rules for backends we've never heard of. If you have the XPU plugin installed, then we should allow such rules to be registered. You could also add `xpu` here https://github.com/jax-ml/jax/blob/62351585824c5f5bb70009456e4ea4949fad0320/jax/_src/xla_bridge.py#L206 so we allow it even if there's no such plugin installed.

The goal of the check is mostly to catch typos, not to forbid new backends.

**Reply by dionhaefner** [reply]
That makes sense, thanks
